config/extension_import_variables.go

```diff
@@ -48,39 +42,30 @@ func (item ImportVariables) help() (result string) {
 }
 
 func (item *ImportVariables) normalize() {
-	if item.FlattenLevels == nil {
-		value := -1
-		item.FlattenLevels = &value
-	}
 	if len(item.NestedObjects) == 0 {
 		// By default, we load the variables at the root
 		item.NestedObjects = []string{""}
 	}
 }
 
-func (item *ImportVariables) loadVariablesFromFile(file string, currentVariables map[string]interface{}) (map[string]interface{}, error) {
+func (item *ImportVariables) loadVariablesFromFile(file string) error {
 	item.logger().Debug("Importing ", file)
 	vars, err := item.options().LoadVariablesFromFile(file)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return item.loadVariables(currentVariables, vars, options.VarFile)
+	item.loadVariables(vars, options.VarFile)
+	return nil
 }
 
-func (item *ImportVariables) loadVariables(currentVariables map[string]interface{}, newVariables map[string]interface{}, source options.VariableSource) (map[string]interface{}, error) {
+func (item *ImportVariables) loadVariables(newVariables map[string]interface{}, source options.VariableSource) {
 	for _, nested := range item.NestedObjects {
 		imported := newVariables
 		if nested != "" {
 			imported = map[string]interface{}{nested: imported}
 		}
 		item.options().ImportVariablesMap(imported, source)
-		flattenedVariables := flatten(imported, "", *item.FlattenLevels)
```

Comment:
I don't really remember, what was the main thing forcing us to flatten the variables?

Reply:
Terraform < 0.12 does not deal with nested structures correctly

Reply:
Terraform 0.11 didn't support accessing nested variables